quarkus-debezium-parent/quarkus-debezium-mysql-parent/deployment/src/main/java/io/quarkus/deployment/MySqlEngineProcessor.java

```diff
@@ -0,0 +1,71 @@
+ */
+package io.quarkus.deployment;
+
+import io.debezium.connector.mysql.Module;
+import io.debezium.connector.mysql.MySqlConnector;
+import io.debezium.connector.mysql.MySqlConnectorTask;
+import io.debezium.connector.mysql.MySqlSourceInfoStructMaker;
+import io.debezium.connector.mysql.snapshot.lock.DefaultSnapshotLock;
+import io.debezium.connector.mysql.snapshot.lock.ExtendedSnapshotLock;
+import io.debezium.connector.mysql.snapshot.lock.MinimalPerconaNoTableLocksSnapshotLock;
+import io.debezium.connector.mysql.snapshot.lock.MinimalPerconaSnapshotLock;
+import io.debezium.connector.mysql.snapshot.lock.MinimalSnapshotLock;
+import io.debezium.connector.mysql.snapshot.lock.NoneSnapshotLock;
+import io.debezium.connector.mysql.snapshot.query.SelectAllSnapshotQuery;
+import io.debezium.relational.history.SchemaHistory;
+import io.debezium.storage.kafka.history.KafkaSchemaHistory;
+import io.quarkus.debezium.agroal.configuration.AgroalDatasourceConfiguration;
+import io.quarkus.debezium.deployment.QuarkusEngineProcessor;
+import io.quarkus.debezium.deployment.items.DebeziumConnectorBuildItem;
+import io.quarkus.debezium.deployment.items.DebeziumExtensionNameBuildItem;
+import io.quarkus.debezium.engine.MySqlEngineProducer;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.pkg.steps.NativeOrNativeSourcesBuild;
+
+public class MySqlEngineProcessor implements QuarkusEngineProcessor<AgroalDatasourceConfiguration> {
+
+    private static final String MYSQL = Module.name();
+
+    @BuildStep
+    @Override
+    public DebeziumExtensionNameBuildItem debeziumExtensionNameBuildItem() {
+        return new DebeziumExtensionNameBuildItem(MYSQL);
+    }
+
+    @BuildStep
+    @Override
+    public DebeziumConnectorBuildItem engine() {
+        return new DebeziumConnectorBuildItem(MYSQL, MySqlEngineProducer.class);
+    }
+
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
+    @Override
+    public void registerClassesThatAreLoadedThroughReflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClassBuildItemBuildProducer) {
+        reflectiveClassBuildItemBuildProducer.produce(ReflectiveClassBuildItem.builder(
+                SchemaHistory.class,
+                KafkaSchemaHistory.class,
+                MySqlConnector.class,
+                MySqlSourceInfoStructMaker.class,
+                MySqlConnectorTask.class,
+                DefaultSnapshotLock.class,
+                ExtendedSnapshotLock.class,
+                MinimalPerconaNoTableLocksSnapshotLock.class,
+                MinimalPerconaSnapshotLock.class,
+                MinimalSnapshotLock.class,
+                NoneSnapshotLock.class,
+                SelectAllSnapshotQuery.class)
+                .reason(getClass().getName())
+                .build());
+    }
+
```

Comment:
What’s currently missing is only the `devservice` for MySQL. For SQL Server, there's nothing specific needed to make the database CDC-ready, but I don't think that's the case for MySQL.

It would be nice to have an empty (without any  example database data) CDC-ready database/schema (maybe the default one), but I am not sure if we already have that. Can we still use `example-mysql` n any case? I have some doubts, especially if a user expects a devservice with example data. When I started developing the extensions, I used the example one for PostgreSQL, but I’ve since changed my mind (😬 ). WDYT?